src/pass/storage_rewrite.cc

```diff
@@ -824,9 +838,7 @@ class StoragePlanRewriter : public IRMutator {
       for (auto it = sym_free_list_.begin();
            it != sym_free_list_.end(); ++it) {
         StorageEntry* e = *it;
-        if (e->attach_scope_ != attach_scope) continue;
-        if (e->scope != scope) continue;
-        if (e->elem_type != op->type.element_of()) continue;
+        if (!CanReuse(e, op, attach_scope, scope)) continue;
```

Comment:
This piece of logic is for dynamic allocation(memory size not known at compile time). We can keep them as it is, by restricting the element size for now.

Reply:
ok